src/PerformanceTests/Tests/Categories/Base.cs

```diff
@@ -109,12 +109,7 @@ static void LaunchAndWait(Permutation permutation)
 
             using (var p = Process.Start(pi))
             {
-                var maxDurationInSeconds = 150;
-                if (!p.WaitForExit(maxDurationInSeconds * 1000))
-                {
-                    p.Kill();
-                    Assert.Fail($"Killed process because execution took more then {maxDurationInSeconds} seconds.");
-                }
+                p.WaitForExit();
```

Comment:
It's a console app so I don't know if `CloseMainWindow` has any effect on that. This routine could potentially also be but in some kind of 'retry' loop with short incrementing sleeps and eventually do throw.